state/mongodb/mongodb_query.go

```diff
@@ -0,0 +1,176 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package mongodb
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/dapr/components-contrib/state"
+	"github.com/dapr/components-contrib/state/query"
+)
+
+type Query struct {
+	query  string
+	filter interface{}
+	opts   *options.FindOptions
+}
+
+func (q *Query) VisitEQ(f *query.EQ) (string, error) {
+	// { <key>: <val> }
+	return fmt.Sprintf("{ %q: %q }", f.Key, f.Val), nil
+}
+
+func (q *Query) VisitIN(f *query.IN) (string, error) {
+	// { $in: [ <val1>, <val2>, ... , <valN> ] }
+	if len(f.Vals) == 0 {
+		return "", fmt.Errorf("empty IN operator for key %q", f.Key)
+	}
+	str := fmt.Sprintf(`{ %q: { "$in": [ %q`, f.Key, f.Vals[0])
+	for _, v := range f.Vals[1:] {
+		str += fmt.Sprintf(", %q", v)
+	}
+	str += " ] } }"
+
+	return str, nil
+}
+
+func (q *Query) visitFilters(op string, filters []query.Filter) (string, error) {
```

Comment:
Also, instead of returning string, it could return interface{}, so for databases like MongoDb, it would already be the struct that the filter expects.

Reply:
This function returns string because it generates the query, which is string